src/seismometer/controls/thresholds.py

```diff
@@ -0,0 +1,154 @@
+        self.sliders = {}
+        for name, val in self.value.items():
+            sub_slider = FloatSlider(
+                value=val,
+                min=0,
+                max=1.0,
+                step=0.01,
+                description=name,
+                disabled=False,
+                continuous_update=False,
+                orientation="horizontal",
+                readout=True,
+                readout_format=".2f",
+                style=WIDE_LABEL_STYLE,
+            )
+            self.sliders[name] = sub_slider
+            sub_slider.observe(self._on_slider_change, "value")
+
+        self.value_update_in_progress = False
+
+        super().__init__(children=list(self.sliders.values()), layout=Layout(width="max-content", min_width="300px"))
+        self.observe(self._on_value_change, "value")
+        self._disabled = False
+
+    @property
+    def disabled(self) -> bool:
+        return self._disabled
+
+    @disabled.setter
+    def disabled(self, disabled: bool):
+        self._disabled = disabled
+        for slider in self.sliders.values():
+            slider.disabled = disabled
+
+    def _on_slider_change(self, change=None):
+        """Slider has changed, update the value tuple"""
+        if self.value_update_in_progress:
+            return
+        if change:
+            if (slider := change["owner"]) in self.sliders.values():
+                self.value[slider.description] = slider.value
+
+    def _on_value_change(self, change=None):
+        """Bubble up changes to sliders"""
+        self.value_update_in_progress = True
+        if change:
+            if val := change["new"]:
+                for key, value in val.items():
+                    self.sliders[key].value = value
+        self.value_update_in_progress = False
+
+
+class MonotonicProbabilitySliderListWidget(ProbabilitySliderListWidget):
+    """
+    Vertical list of sliders, bounded between 0 and 1 with a step of 0.01.
+    Monotonicity is maintained between the sliders so they are always ascending or descending in value.
+    """
+
+    def __init__(self, names: tuple[str], value: tuple[int] = None, ascending: bool = True):
+        """A vertical list of sliders, bounded between 0 and 1 with a step of 0.01.
+
+        Parameters
+        ----------
+        names : tuple[str]
+            Slider names
+        value : Optional[tuple[str]], optional
+            Slider start values, by default None, starts all sliders at zero.
+        ascending : bool, optional = True
+            Forces sliders to be ascending, else decreasing.
+            If initial values are not sorted, raise an ValueError.
+        """
+        if len(names) > 6:
+            raise ValueError("MonotonicProbabilitySliderListWidget only supports up to 6 sliders")
+
+        super().__init__(names, value)
+
+        if tuple(sorted(self.value.values(), reverse=not ascending)) != tuple(self.value.values()):
```

Comment:
Also, wouldn't it be more efficient to do a linear pass through the tuple rather than sorting and comparing?

Reply:
Yes, but this is a pretty rare code path, with only a few values to sort, so not that slow. that being said, I did do this change mostly because I don't need to call sorted and have the casts to tuple. 

FINC